Include path of boost headers

Hi, I encountered an error in compiling GM2Calc (latest master branch) on my MacOS 14.

```
GM2Calc_src/src/slhaea.h:25:10: fatal error: 'boost/algorithm/string/classification.hpp' file not found
#include <boost/algorithm/string/classification.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
```

I suppose this is because the Boost path is introduced privately in the commit ec6004445424dc1596189e2143b2dd62312c6b2e. In fact, the following patch solves this problem.

```diff
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -40,9 +40,9 @@ target_include_directories(gm2calc
   PUBLIC
     "${PROJECT_SOURCE_DIR}/include"
     "${EIGEN3_INCLUDE_DIR}"
+    "${Boost_INCLUDE_DIRS}"
   PRIVATE
     "${CMAKE_CURRENT_SOURCE_DIR}"
-    "${Boost_INCLUDE_DIRS}"
```

Are there particular reason to load it privately? or otherwise I propose to apply this patch.

Best,
Sho

It might be because usually `-I` for Eigen3 and Boost are coincide. In my environment they (somehow) locates at
```
/usr/local/include/eigen3/Eigen/Core # (-> -I/u/l/i/eigen3
/usr/local/include/boost/regex.h     # (-> -I/u/l/i
```
but it is fair to keep it private as it is well-tested.

This is a full log, starting from git clone.
```console
$ git clone https://github.com/GM2Calc/GM2Calc.git GM2Calc_src
$ cd GM2Calc
$ cmake ../GM2Calc_src
(snip)
-- Found Boost: /usr/local/lib/cmake/Boost-1.78.0/BoostConfig.cmake (found suitable version "1.78.0", minimum required is "1.37.0")
-- Found Eigen3: /usr/local/include/eigen3 (Required is at least version "3.1")
(snip)

$ make VERBOSE=1
(snip)
[ 43%] Building CXX object src/CMakeFiles/gm2calc.x.dir/gm2calc.cpp.o
cd /Users/misho/HEPcodes/GM2Calc/src && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/misho/HEPcodes/GM2Calc/src -I/Users/misho/HEPcodes/GM2Calc_src/include -I/usr/local/include/eigen3 -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.14 -std=gnu++14 -MD -MT src/CMakeFiles/gm2calc.x.dir/gm2calc.cpp.o -MF CMakeFiles/gm2calc.x.dir/gm2calc.cpp.o.d -o CMakeFiles/gm2calc.x.dir/gm2calc.cpp.o -c /Users/misho/HEPcodes/GM2Calc_src/src/gm2calc.cpp
In file included from /Users/misho/HEPcodes/GM2Calc_src/src/gm2calc.cpp:24:
In file included from /Users/misho/HEPcodes/GM2Calc_src/include/gm2calc/MSSMNoFV_onshell.hpp:22:
In file included from /Users/misho/HEPcodes/GM2Calc_src/include/gm2calc/MSSMNoFV_onshell_mass_eigenstates.hpp:32:
In file included from /Users/misho/HEPcodes/GM2Calc_src/include/gm2calc/MSSMNoFV_onshell_soft_parameters.hpp:22:
In file included from /Users/misho/HEPcodes/GM2Calc_src/include/gm2calc/MSSMNoFV_onshell_susy_parameters.hpp:24:
In file included from /usr/local/include/eigen3/Eigen/Core:15:
/usr/local/include/eigen3/Eigen/src/Core/util/DisableStupidWarnings.h:48:38: warning: unknown warning group '-Wimplicit-int-float-conversion', ignored [-Wunknown-warning-option]
    #pragma clang diagnostic ignored "-Wimplicit-int-float-conversion"
                                     ^
In file included from /Users/misho/HEPcodes/GM2Calc_src/src/gm2calc.cpp:32:
In file included from /Users/misho/HEPcodes/GM2Calc_src/src/gm2_slha_io.hpp:24:
/Users/misho/HEPcodes/GM2Calc_src/src/slhaea.h:25:10: fatal error: 'boost/algorithm/string/classification.hpp' file not found
#include <boost/algorithm/string/classification.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
make[2]: *** [src/CMakeFiles/gm2calc.x.dir/gm2calc.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/gm2calc.x.dir/all] Error 2
make: *** [all] Error 2
```

Best,
Sho

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -56,7 +56,11 @@ install(
 # GM2Calc main executable
 add_executable(gm2calc.x gm2calc.cpp)
 target_link_libraries(gm2calc.x PRIVATE GM2Calc::GM2Calc)
-target_include_directories(gm2calc.x PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
+target_include_directories(gm2calc.x
+  PRIVATE
+    "${CMAKE_CURRENT_BINARY_DIR}"
+    "${Boost_INCLUDE_DIRS}"
+)
 
 install(
   TARGETS gm2calc.x
@@ -70,7 +74,11 @@ if(Mathematica_MathLink_FOUND)
     gm2calc.tm)
   target_link_libraries(gm2calc.mx PRIVATE GM2Calc::GM2Calc)
   target_include_directories(gm2calc.mx
-    PRIVATE ${Mathematica_MathLink_INCLUDE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+    PRIVATE
+      "${Mathematica_MathLink_INCLUDE_DIR}"
+      "${CMAKE_CURRENT_SOURCE_DIR}"
+      "${Boost_INCLUDE_DIRS}"
+  )
   install(
     TARGETS gm2calc.mx
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}

